### DISCUSSION TITLE
Automatically Move to Next Iteration

### DISCUSSION THREAD

**Discussion by bbernays** [community/community#16222] [Projects and Issues] [answered]
At the end of an iteration it would be great if incomplete issues automatically got moved to the next iteration. Without this issues either get lost because filters only show current iteration or we have to manually move the issues over.

In the short term this functionality could be implemented with a GitHub action that was invoked at the start/end of an iteration.

**Comment by stephent** [comment]
Surprised this hasn't had more comment - it's too easy to lose sight of incomplete issues from previous iterations at the moment, in my experience. It would be great to be able to decide what to do with outstanding issues when an iteration 'closes'.

**Comment by jaclynjessup** [comment]
Would also really like this feature and would like to know that it was carried over from the previous iteration. I currently have to keep track of this manually and within a separate spreadsheet.

**Comment by gigpiged** [comment]
Also this, every two weeks we have to move any open tasks not 'done' in the past iteration, it should do this automatically.

**Comment by Flaflo** [comment]
That this is not answered yet by GH is very sad imho.

**Comment by ghost** [comment]
Assuming I have an iteration field called "Sprint"

This would be very useful. If the workflows feature was a bit more advanced, it could be easily done:
- When `Sprint X` reaches due date, change all associated issues to `Sprint X+1` (or `@next` iteration).

Would also be useful:
- When `Sprint X` reaches due date, associate a specific label to all associated issues. I would choose a label like `Carried Over From Sprint X` to indicate it was carried over);
- When `Sprint X` reaches due date, change custom field to "Y" value on all associated issues.

**Comment by blombard** [comment]
Since Github has other fish to fry, I made my own Github action: https://github.com/marketplace/actions/move-to-next-iteration

**Reply by Flaflo** [reply]
Thank you very much! <3

**Comment by henrystaubyn** [comment]
Agreed this would be a very useful feature.  One more reason to move back to Jira atm...

**Comment by dalerocks** [comment]
Also agreed this would help avoid Issues slipping through the cracks.

**Comment by iamkoby** [comment]
Can't believe this is not built in

**Comment by mepps** [comment]
Adding a +1 for this feature request! Or a view that shows all issues which haven't rolled over to make it easy to select them.

**Reply by diaristadeveloper** [reply]
Well, I dunno if move to next iteration is helpful to all scenarios.
Some tasks might be de-prioritized if not executed in current iteration, for example.
Also, in sprint review you have oportunity to choose wether to move to next or return it to backlog.

**Comment by oemebamo** [comment]
Adding a +1 for this feature request.

Now moving tickets one by one.

**Comment by mason-rogers** [comment]
+1 - thanks to @blombard for making a github action in the meantime - this should be native...

**Comment by sabpamdev1** [comment]
Its September 2024 and I would upvote for this feature to be native.

**Comment by jusvit** [comment]
This is embarrassing for project planning

**Comment by katerynafiroozi** [comment]
will there be an update for node20?

**Reply by blombard** [reply]
If you are talking about the Github action, yes it's been updated in https://github.com/blombard/move-to-next-iteration/releases/tag/v0.6.0

**Comment by beingminimal** [comment]
+1

**Comment by dagan** [comment]
FWIW, you can show all issues that (1) have an iteration assigned _and_ (2) the assigned iteration is _either_ current iteration or a previous one with `iteration:<@next`. You can also set up a planning view that uses a range like `iteration:@previous-1..@next+1` to span the previous two iterations, current iteration, and next two iterations.

I agree that automatically moving the iteration forward should be an available feature, though.

**Comment by afern247** [comment]
I was planning to use Github projects for spring planning but this is a real deal breaker, I don't want to focus on moving stuff to sprints, I want to focus on shipping products.

+1 Please implement this

**Reply by blombard** [reply]
FYI this is already possible with [this Github action](https://github.com/marketplace/actions/move-to-next-iteration). (If that's the only thing holding you back from using Github project)

**Reply by afern247** [reply]
@bbernays I saw that github action, thanks, I don't want hacky things 🙂

**Reply by kutsan** [reply]
I think this feature might not even come directly. Supporting basic features and allowing it to be extended via API sounds like a valid roadmap to me. So, I don't think using an action workflow for this is a hack.

**Reply by afern247** [reply]
not a hack? The you are thinking as an engineer that likes to build stuff not as a product owner. Not a hack means things work out of the box, no setup required it's ready to be used as a project manager with the tools needed for non tech people to do stuff on the UI and focus on productivity. Yes, it is a hack... imagine jira having to do that or notion...

**Reply by kutsan** [reply]
I'm not saying I don't want this to come; in fact, it would be very good if it did, but since the last time I checked, I don't think GitHub is heading in a product manager-first direction like Jira or Notion. If GitHub Projects is following such a roadmap, I think that's great, but I consider GitHub to be a developer-first platform, and using GitHub Actions for this use case doesn't seem unreasonable to me. Furthermore, this can also be said for other things that Jira or Notion support but GitHub Projects doesn't. If specific use cases can be handled with community plugins, I prefer extensibility over having them built-in since we don't need to wait for them to be implemented.

Also, don't misunderstand me. I also want this feature to be implemented, that's why I'm here. However, product managers can set this up even now with the help of a developer and this might be enough.

**Comment by jonathanva** [comment]
Sad that this ticket is still open after 3 year. Feels like a basic feature. There is indeed an workaround action, but within the company we cannot use unverified actions. So the only solution would be to implement it ourselves.

I think a lot of enterprise customers would benefit from this feature.

**Comment by steffengoertz** [comment]
I don't have the impression that anyone is still working on Projects at Github. Last product update is from over a year ago, looks quite abandoned to me tbh.

**Comment by tlabeeuw** [comment]
I wrote an entire Github Actions to re-implement how https://www.pivotaltracker.com/ does XP programming and calculating iterations and running velocity to move stories appropriate. Will maybe release publically but we ended up writing this all by hand.

**Comment by azenMatt** [comment, marked as answer]
This isn't fully automated, but we released an update here - https://github.blog/changelog/2026-03-05-hierarchy-view-improvements-and-file-uploads-in-issue-forms/#move-to-next-iteration-in-bulk - that will make it much quicker to move these over

**Reply by kutsan** [reply]
Are there any plans to automate this in the future or should we start using GitHub Actions instead?

**Reply by phifuh** [reply]
bulk edit is nice, but its about work smarter, not harder
